code/modules/projectiles/guns/energy/temperature.dm

```diff
@@ -1,14 +1,169 @@
+		if("High")		powercostcolor = "orange"
+		if("Medium")	powercostcolor = "green"
+		else			powercostcolor = "blue"
+	if(target_temperature != temperature)
+		var/difference = abs(target_temperature - temperature)
+		if(difference >= (10 + 40*emagged)) //so emagged temp guns adjust their temperature much more quickly
+			if(target_temperature < temperature)
+				temperature -= (10 + 40*emagged)
+			else
+				temperature += (10 + 40*emagged)
+		else
+			temperature = target_temperature
+		update_icon()
+
+		if (istype(loc, /mob/living/carbon))
+			var /mob/living/carbon/M = loc
+			if (src == M.machine)
+				update_dat()
+				M << browse("<TITLE>Temperature Gun Configuration</TITLE><HR>[dat]", "window=tempgun;size=510x102")
+
+
+/obj/item/weapon/gun/energy/temperature/proc/update_dat()
+	dat = ""
+	dat += "Current output temperature: "
+	if(temperature > 500)
+		dat += "<FONT color=red><B>[temperature]</B> ([round(temperature-T0C)]&deg;C) ([round(temperature*1.8-459.67)]&deg;F) </FONT>"
+		dat += "<FONT color=red><B>SEARING!!</B></FONT>"
+	else if(temperature > (T0C + 50))
+		dat += "<FONT color=red><B>[temperature]</B> ([round(temperature-T0C)]&deg;C) ([round(temperature*1.8-459.67)]&deg;F)</FONT>"
+	else if(temperature > (T0C - 50))
+		dat += "<FONT color=black><B>[temperature]</B> ([round(temperature-T0C)]&deg;C) ([round(temperature*1.8-459.67)]&deg;F)</FONT>"
+	else
+		dat += "<FONT color=blue><B>[temperature]</B> ([round(temperature-T0C)]&deg;C) ([round(temperature*1.8-459.67)]&deg;F)</FONT>"
+	dat += "<BR>"
+	dat += "Target output temperature: "	//might be string idiocy, but at least it's easy to read
+	dat += "<A href='?src=\ref[src];temp=-100'>-</A> "
+	dat += "<A href='?src=\ref[src];temp=-10'>-</A> "
+	dat += "<A href='?src=\ref[src];temp=-1'>-</A> "
+	dat += "[target_temperature] "
+	dat += "<A href='?src=\ref[src];temp=1'>+</A> "
+	dat += "<A href='?src=\ref[src];temp=10'>+</A> "
+	dat += "<A href='?src=\ref[src];temp=100'>+</A>"
+	dat += "<BR>"
+	dat += "Power cost: "
+	dat += "<FONT color=[powercostcolor]><B>[powercost]</B></FONT>"
+/obj/item/weapon/gun/energy/temperature/proc/update_temperature()
+	switch(temperature)
+		if(501 to INFINITY)
+			item_state = "tempgun_8"
+		if(400 to 500)
+			item_state = "tempgun_7"
+		if(360 to 400)
+			item_state = "tempgun_6"
+		if(335 to 360)
+			item_state = "tempgun_5"
+		if(295 to 335)
+			item_state = "tempgun_4"
+		if(260 to 295)
+			item_state = "tempgun_3"
+		if(200 to 260)
+			item_state = "tempgun_2"
+		if(120 to 260)
+			item_state = "tempgun_1"
+		if(-INFINITY to 120)
+			item_state = "tempgun_0"
+	icon_state = item_state
+/obj/item/weapon/gun/energy/temperature/update_icon()
+	overlays = 0
+	update_temperature()
+
+/obj/item/weapon/gun/energy/temperature/ultra
+	name = "ultra temperature gun"
+	desc = "A gun that changes the body temperature of its targets to any temperature. ANY. TEMPERATURE.."
+
+/obj/item/weapon/gun/energy/temperature/ultra/Topic(href, href_list)
```

Comment:
I don't like this implementation.

Instead, please make the min and max temperatures vars of the gun, give both a default minimum of zero, the plain gun a max of 500, and the ultra a max of like 1e10.